### PR TITLE
[Fix] Pin icon should not be visible to users without pin permission (revised)

### LINKED ISSUE #519: [Feature Request] Pin icon should not be visible to users without pin permissions
Pin icon is not visible to users without pin permissions in the main Rocket.Chat application. This should be implemented in EmbeddedChat as well.  

**Acceptance Criteria**

- [ ] Pin icon to be not visible to users not having pin permissions.

ss_144

ss_145

### PR DESCRIPTION
**Pull request by thesynthax**
# Brief Title
Pin icon should not be visible to users without the pin permission like in main Rocket.Chat application.
Check #518, #519 and #520 for complete details

## Acceptance Criteria fulfillment

- [x] Pin Icon is now not visible to users not having Pin Permissions.

Fixes #519

## Video/Screenshots

ss_146

P.S: This PR is a revision of #520 after #518 is merged. Following this, #520 will be discarded.

### CODE REVIEW

**Comment by thesynthax** [comment]
@Spiral-Memory @sidmohanty11 please have a look.

File: packages/react/src/components/ChatHeader/ChatHeader.js [hunk 1]
```diff
@@ -155,29 +155,44 @@ const ChatHeader = ({
...
-        const userRoles = channelRolesRes.roles
+        const channelMemberRoles = channelRolesRes.roles
           .filter((chRole) => chRole.u?._id === authenticatedUserId)
           .flatMap((chRole) => chRole.roles);
 
-        const canSendMsg =
-          userRoles.length > 0 &&
-          postMsgRoles.some((role) => userRoles.includes(role));
-        setCanSendMsg(canSendMsg);
+        const concatenatedRoles = channelMemberRoles.concat(
```
Comment (Spiral-Memory) [inline packages/react/src/components/ChatHeader/ChatHeader.js:181]: You can use shorthand here, destructure and combine. I have raised a PR feat: Addition of chatinfo to display status, have a look here..

File: packages/react/src/components/Message/Message.js [hunk 2]
```diff
@@ -235,6 +235,7 @@ const Message = ({
           {!message.t && showToolbox ? (
             <MessageToolbox
               message={message}
+              RCInstance={RCInstance}
```
Comment (Spiral-Memory) [inline packages/react/src/components/Message/Message.js:238]: Why is this added ?

Comment (devanshkansagra) [reply packages/react/src/components/Message/Message.js:238]: Hey @Spiral-Memory the adding of RCInstance as a prop was unnecessary, after removing it I have seen no change

File: packages/react/src/components/ChatHeader/ChatHeader.js (hunk 1 above)
Comment (devanshkansagra) [reply packages/react/src/components/ChatHeader/ChatHeader.js:181]: Get @Spiral-Memory, if we destructure and combine, the users are unable to login

**Comment by Spiral-Memory** [comment]
Closing this PR, as the pin permission display is implemented recently
